src/components/ErrorBoundary/utils.tsx

```diff
@@ -0,0 +1,79 @@
+export async function collectDiagnosticsData(error: Error) {
+    return await Promise.all([getUiVersion(), getBackendVersion()]).then(
+        ([uiVersion, backendVersion]) => {
+            return {
+                location: window.location.href,
+                userAgent: navigator.userAgent,
+                error: {
+                    message: prepareErrorMessage(error),
+                    stack: prepareErrorStack(error.stack, {trim: true, maxLength: 10}),
+                },
+                uiVersion,
+                backendVersion,
+            };
+        },
+    );
+}
+
+export type DiagnosticsData = Awaited<ReturnType<typeof collectDiagnosticsData>>;
+
+// Error could happen on app init, while modules not inited
+// Import in try catch blocks to prevent any errors with error data collection
+async function getUiVersion() {
+    try {
+        const packageJson = await import('../../../package.json');
```

Comment:
not sure if package.json is really available in frontend production bundle
not sure though

Reply:
<img width="806" alt="image" src="https://github.com/user-attachments/assets/e6708d54-7197-41ac-a427-080bc2c67c30" />

hmm I see its integrated in bundle

Reply:
could you try something like 

```
const packageJson = require('./package.json');

module.exports = {
    webpack: (config, env) => {
        // Your existing webpack config modifications...
        
        // Add DefinePlugin to expose just the version
        config.plugins.push(
            new webpack.DefinePlugin({
                'process.env.UI_VERSION': JSON.stringify(packageJson.version)
            })
        );
        
        return config;
    },
    // Rest of your config...
};
```

in config-overrides.js


the idea of including the whole package json seems not the cleanest approach (just for me though)

Reply:
Thanks, no `package.json` in bundle anymore

![Screenshot 2025-02-26 at 19 25 06](https://github.com/user-attachments/assets/4c0da32b-ea51-40df-bfda-df142cf30c50)